### PR TITLE
330 - Implement IDS Contextual Action Panel

### LINKED ISSUE #330: IdsContextualActionPanel: New Web Component
Add a new `IdsContextualActionPanel` (ids-contextual-action-panel) component MVP to the WebComponent repo based on https://main-enterprise.demo.design.infor.com/components/contextualactionpanel

Think it should work like the modal possibly inheriting from it. Just change the animation and the toolbar and styling. Should be possible to set the content with the shadowDOM/slots not using a buttons setting like before

**Phase 1 Features (MVP)**
- Animation
- Toolbar
- Contents
- Events fire
- Modal Breakpoints/fullsize
- use new design https://github.com/infor-design/enterprise/pull/5688

**Phase 2 Features (MVP)**
- Most Anything else 

**Possible Internal Markup**
```html
<ids-contextual-action-panel id="my-cap" aria-labelledby="my-cap-title">
  <ids-text slot="title" font-size="24" type="h2" id="my-cap-title">Active IDS CAP</ids-text>
  <ids-toolbar slot="toolbar">
  </ids-toolbar>
  <div slot="contents">
     <!-- contents -->
  </div>
</ids-contextual-action-panel>
```

**Suggested Dependencies**
- ids-modal

**Design**
https://main-enterprise.demo.design.infor.com/components/line/example-index.html

**Events**
- beforeopen, open, afteropen
- beforeclose, close, afterclose

Close are vetoable
https://github.com/infor-design/enterprise-wc/blob/main/src/components/ids-tag/ids-tag.js#L225-L233

**Other**
Pass the following checklist ordered in priority (some may be left for later) https://github.com/infor-design/enterprise-wc/blob/main/doc/CHECKLIST.md

### PR DESCRIPTION
**Pull request by EdwardCoyle**
**Explain the details for making this change. What existing problem does the pull request solve?**
This PR adds the IDS Contextual Action Panel component
- Extends Modal component
- Implements a toolbar area
- The Toolbar's contents are connected to the `onButtonClick` callback provided by the Modal
- Makes adjustments to Popup component that fix placement coordinates to account for components nested inside positioned parent elements.
- Adds docs/types

**Related github/jira issue (required)**:
Closes #330

**Steps necessary to review your pull request (required)**:
- Pull, build, run
- Open http://localhost:4300/ids-contextual-action-panel and click the trigger to open
- See that all form controls and buttons inside the CAP work (close/save are wired to hide the CAP)

**Included in this Pull Request**:
- [x] An e2e or functional test for the bug or feature.
- [x] A note to the change log.

### CODE REVIEW

File: demos/ids-contextual-action-panel/example.html [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+      <ids-button id="btn-save" icon="save" no-padding>
+        <ids-text font-weight="bold">Save</ids-text>
+      </ids-button>
+      <ids-separator vertical="true"></ids-separator>
+      <ids-button id="btn-close" icon="close" no-padding>
+        <ids-text font-weight="bold">Close</ids-text>
+      </ids-button>
+    </ids-toolbar-section>
+
+    <!--
```
Comment (tmcconechy) [inline demos/ids-contextual-action-panel/example.html:17]: Any reason to keep?

File: src/components/ids-contextual-action-panel/ids-contextual-action-panel.js [hunk 2]
```diff
@@ -0,0 +1,89 @@
+import {
+  customElement,
+  scss,
+} from '../../core/ids-element';
+
+import IdsModal from '../ids-modal';
+import '../ids-toolbar';
```
Comment (tmcconechy) [inline src/components/ids-contextual-action-panel/ids-contextual-action-panel.js:7]: Should it import something? import idsToolbar from  '../ids-toolbar';

File: src/components/ids-contextual-action-panel/ids-contextual-action-panel.js [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+
+import styles from './ids-contextual-action-panel.scss';
+
+/**
+ * IDS Contextual Action Panel
+ * @type {IdsContextualActionPanel}
+ * @inherits IdsModal
+ * @mixes IdsEventsMixin
```
Comment (tmcconechy) [inline src/components/ids-contextual-action-panel/ids-contextual-action-panel.js:15]: Doesnt directly mix so not sure if needed?

**Review by tmcconechy — COMMENTED** [review]
Looks awesome 👍🏻  
- Only a couple simple comments
- In terms of functionality one thing that stands out is when you click the CAP buttons the "ripple effect" is much darker before. Maybe needs a lighter style like https://main-enterprise.demo.design.infor.com/components/contextualactionpanel/example-index.html
- i also noticed on about/cap ect. that clicking a button to close has a short delay before actually closing - might be an issue in the future?

File: src/components/ids-contextual-action-panel/ids-contextual-action-panel.js (hunk 2 above)
Comment (EdwardCoyle) [reply src/components/ids-contextual-action-panel/ids-contextual-action-panel.js:7]: I'm just depending on it here for the template.  I'm not using the named export (not invoking with `new`) so didn't use it.

**Comment by EdwardCoyle** [comment]
@tmcconechy I addressed all the comments here except for this one:

> i also noticed on about/cap ect. that clicking a button to close has a short delay before actually closing - might be an issue in the future?

**Comment by tmcconechy** [comment]
ok, maybe we might end up shortening that delay a bit as it might "slow down" the workflow. But we can wait for further feedback. I wasn't sure on it

**Review by tmcconechy — APPROVED** [review]

**Comment by EdwardCoyle** [comment]
@tmcconechy Just rebased this against main, shouldn't have any changes

**Review by tmcconechy — APPROVED** [review]

**Review by tmcconechy — APPROVED** [review]

**Review by jmacaluso711 — APPROVED** [review]
Looks good! 👍

**Pull request merged**